app/frontend/src/components/Comments/NewComment.tsx

```diff
@@ -0,0 +1,61 @@
+import { Box, Button, Grid, Link, makeStyles } from "@material-ui/core";
+import React, { useState } from "react";
+
+import Markdown from "../../components/Markdown";
+import TextField from "../../components/TextField";
+
+const useStyles = makeStyles({
+  reverseAlignment: {
+    textAlign: "end",
+  },
+});
+
+interface NewCommentProps {
+  onComment: (comment: string) => Promise<void>;
+}
+
+export default function NewComment({ onComment }: NewCommentProps) {
+  const classes = useStyles();
+  const [preview, setPreview] = useState(false);
+  const [comment, setComment] = useState("");
+
+  const handleSubmit = async () => {
+    await onComment(comment);
+    setComment("");
+  };
+
+  return (
+    <>
+      <p>Write a comment:</p>
+      <Grid container spacing={2}>
+        <Grid item xs={12} md={preview ? 6 : 12}>
+          <TextField
+            label="Text field"
+            rowsMax={5}
+            multiline
+            fullWidth
+            onChange={(e) => setComment(e.target.value)}
```

Comment:
We should wrap this in a form and do it with `react-hook-form` like in other places, since it optimises for state change like this that otherwise would make this re-render on every keystroke and cause performance issues.

Reply:
@aapeliv if you make a note on #543 that this needs to be converted to react query and use react-hook-form, you can leave it for us to do